test_bst.py

```diff
@@ -48,13 +48,13 @@ def test_depth(full_tree, empty_tree):
 
 
 def test_balance(full_tree, empty_tree):
-    assert full_tree.balance() < 0
+    assert full_tree.balance() > 0
     assert empty_tree.balance() == 0
     empty_tree.insert(5)
     assert empty_tree.balance() == 0
     empty_tree.insert(8)
-    assert empty_tree.balance() > 0
+    assert empty_tree.balance() < 0
     empty_tree.insert(4)
     assert empty_tree.balance() == 0
     empty_tree.insert(1)
-    assert empty_tree.balance() < 0
+    assert empty_tree.balance() > 0
```

Comment:
These tests are pretty thin.  I'd like to see you working a bit harder to ensure that things work as you expect them to.  For example, you should be testing the relationship between the `.insert` method and the `.depth` method.  Since depth is updated on insert, you want to ensure that it is working correctly for a variety of scenarios.